main.tf

```diff
@@ -1,3 +1,83 @@
+  validate_service_end_points_lite_standard = ((var.plan == "lite" || var.plan == "standard") && var.service_endpoints != "public") ? tobool("Service endpoint cannot be changed in lite and standard plan. Default is public.") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_private_ip_allowlist_lite_standard = ((var.plan == "lite" || var.plan == "standard") && var.private_ip_allowlist != null) ? tobool("Private ip allowlist cannot be changed in lite and standard plan.") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_hpcs_guid_input = var.skip_iam_authorization_policy == false && var.existing_kms_instance_guid == null ? tobool("A value must be passed for var.existing_kms_instance_guid when creating an instance, var.skip_iam_authorization_policy is false.") : true
+}
+
+resource "ibm_resource_instance" "es_instance" {
+  name              = var.es_name
+  service           = "messagehub"
+  plan              = var.plan
+  location          = var.region
+  resource_group_id = var.resource_group_id
+  tags              = var.tags
+  timeouts {
+    create = var.create_timeout
+    update = var.update_timeout
+    delete = var.delete_timeout
+  }
+
+  parameters = {
+    service-endpoints         = var.service_endpoints
+    private_ip_allowlist      = var.private_ip_allowlist
+    throughput                = var.throughput
+    storage_size              = var.storage_size # Storage capacity cannot be scaled down once instance is created.
+    key_protect_key           = var.kms_key_crn
+    backup_encryption_key_crn = local.kp_backup_crn
+  }
+}
+
+##############################################################################
+# SCHEMA
+##############################################################################
+
+resource "ibm_event_streams_schema" "es_schema" {
+  count                = length(var.schemas) * (var.plan == "enterprise-3nodes-2tb" ? 1 : 0)
+  resource_instance_id = ibm_resource_instance.es_instance.id
+  schema_id            = var.schemas[count.index].schema_id
+  schema               = jsonencode(var.schemas[count.index].schema)
+}
+
+##############################################################################
+# TOPIC
+##############################################################################
+
+resource "ibm_event_streams_topic" "es_topic" {
+  count                = length(var.topics)
+  resource_instance_id = ibm_resource_instance.es_instance.id
+  name                 = var.topics[count.index].name
+  partitions           = var.topics[count.index].partitions
+  config               = var.topics[count.index].config
+}
+
+
+##############################################################################
+# IAM Authorization Policy
+##############################################################################
+
+# Create IAM Authorization Policies to allow postgresql to access kms for the encryption key
+resource "ibm_iam_authorization_policy" "kms_policy" {
```

Comment:
Sync up with @shemau on this to ensure we don't introduce this bug reported here https://github.com/terraform-ibm-modules/terraform-ibm-icd-postgresql/issues/165

Reply:
Added a condition for this as also came across it in the examples.